File: minecraft-app/src/main/java/com/mcal/moddedpe/googleplay/ApkXDownloaderService.java
package com.mcal.moddedpe.googleplay;


import com.google.android.vending.expansion.downloader.impl.DownloaderService;


public class ApkXDownloaderService extends DownloaderService {
    @Override
    public String getPublicKey() {
        return ApkXDownloaderClient.getLicenseKey();
    }

    @Override
    public byte[] getSALT() {
        return ApkXDownloaderClient.SALT;
    }

    public String getNotificationChannelId() {
        return ApkXDownloaderClient.getNotificationChannelId();
    }

    @Override
    public String getAlarmReceiverClassName() {
        return ApkXDownloaderAlarmReceiver.class.getName();
    }
}

File: minecraft-app/src/main/java/com/mcal/moddedpe/googleplay/ApkXDownloaderService.kt
package com.mcal.moddedpe.googleplay

import com.google.android.vending.expansion.downloader.impl.DownloaderService


class ApkXDownloaderService : DownloaderService() {
    override fun getPublicKey(): String {
        return ApkXDownloaderClient.getLicenseKey()
    }

    override fun getSALT(): ByteArray {
        return ApkXDownloaderClient.SALT
    }

    fun getNotificationChannelId(): String {
        return ApkXDownloaderClient.getNotificationChannelId()
    }

    override fun getAlarmReceiverClassName(): String {
        return ApkXDownloaderAlarmReceiver::class.java.name
    }
}
